src/common/install.js

```diff
@@ -2,20 +2,28 @@
 
 var HypothesisChromeExtension = require('./hypothesis-chrome-extension');
 
-var browserExtension = new HypothesisChromeExtension({
-  chromeExtension: chrome.extension,
-  chromeTabs: chrome.tabs,
-  chromeBrowserAction: chrome.browserAction,
-  chromeStorage: chrome.storage,
-  extensionURL: function(path) {
-    return chrome.extension.getURL(path);
-  },
-  isAllowedFileSchemeAccess: function(fn) {
-    return chrome.extension.isAllowedFileSchemeAccess(fn);
-  },
-});
+var browserExtension;
 
-browserExtension.listen(window);
+function init() {
```

Comment:
The code here has been moved into an init function which is also exported to deal with the main drawback of the new approach, which is that you can't mock code that runs during the initial evaluation of a module. See the "Drawbacks" section of https://github.com/hypothesis/client/pull/1061.

I actually think this might be a good thing by encouraging imports which are side-effect free. However, one place where it will be a problem are modules which are entry-points for the app, like this one. There is a relatively simple workaround though which is to wrap the code in an init function which is run immediately in the real app but not during tests.